Tests/SwiftBuildSupportTests/SwiftBuildSystemMessageHandlerTests.swift

```diff
@@ -0,0 +1,355 @@
+
+        for event in events {
+            _ = try messageHandler.emitEvent(event)
+        }
+
+        #expect(observability.hasErrorDiagnostics)
+
+        try expectDiagnostics(observability.diagnostics) { result in
+            result.check(diagnostic: "Simple diagnostic", severity: .error)
+        }
+    }
+
+    @Test
+    func testManyDiagnosticsReported() throws {
+        let (messageHandler, _, observability) = createMessageHandler()
+
+        let events: [SwiftBuildMessage] = [
+            .taskStartedInfo(taskID: 1, taskSignature: "simple-diagnostic"),
+            .diagnosticInfo(
+                locationContext2: .init(taskSignature: "simple-diagnostic"),
+                message: "Simple diagnostic",
+                appendToOutputStream: true
+            ),
+            .taskStartedInfo(taskID: 2, taskSignature: "another-diagnostic"),
+            .taskStartedInfo(taskID: 3, taskSignature: "warning-diagnostic"),
+            .diagnosticInfo(
+                kind: .warning,
+                locationContext2: .init(taskSignature: "warning-diagnostic"),
+                message: "Warning diagnostic",
+                appendToOutputStream: true
+            ),
+            .taskCompleteInfo(taskID: 1, taskSignature: "simple-diagnostic", result: .failed),
+            .diagnosticInfo(
+                kind: .warning,
+                locationContext2: .init(taskSignature: "warning-diagnostic"),
+                message: "Another warning diagnostic",
+                appendToOutputStream: true
+            ),
+            .taskCompleteInfo(taskID: 3, taskSignature: "warning-diagnostic", result: .success),
+            .diagnosticInfo(
+                kind: .note,
+                locationContext2: .init(taskSignature: "another-diagnostic"),
+                message: "Another diagnostic",
+                appendToOutputStream: true
+            ),
+            .taskCompleteInfo(taskID: 2, taskSignature: "another-diagnostic", result: .failed)
+        ]
+
+        for event in events {
+            _ = try messageHandler.emitEvent(event)
+        }
+
+        #expect(observability.hasErrorDiagnostics)
+
+        try expectDiagnostics(observability.diagnostics) { result in
+            result.check(diagnostic: "Simple diagnostic", severity: .error)
+            result.check(diagnostic: "Another diagnostic", severity: .debug)
+            result.check(diagnostic: "Another warning diagnostic", severity: .warning)
+            result.check(diagnostic: "Warning diagnostic", severity: .warning)
+        }
+    }
+
+    @Test
+    func testCompilerOutputDiagnosticsWithoutDuplicatedLogging() throws {
+        let (messageHandler, outputStream, observability) = createMessageHandler()
+
+        let simpleDiagnosticString: String = "[error]: Simple diagnostic\n"
+        let simpleOutputInfo: SwiftBuildMessage = .outputInfo(
+            data: data(simpleDiagnosticString),
+            locationContext: .task(taskID: 1, targetID: 1),
```

Comment:
question: do the values of `taskID` and `targetID` need to match between all test `SwiftBuildMessage` that are in the `events` array?  If so, consider storing the values in a `let` and using said variable name in various places to make the test more readable.